docs/architecture/adr-104-out-of-band-state-sync.md

```diff
@@ -271,68 +330,125 @@ snapshot_dump_path=""
+
+Applications generate snapshots in fixed time intervals. The application itself measures the time passed since the last snapshot,
+and CometBFT has no role in instructing the application when to take snapshots. 
+
+The State sync reactor currently retrieves a list of snapshots from a peer, who obtains these snapshots via ABCI (`RequestListSnapshots`). 
+
+Applications can thus themselves be in charge of dumping the snapshots into a given file format, the same way they generate snapshots.
+ If `auto_snapshot_dump` is true, 
+CometBFT instructs the application to export the snapshots periodically.
+
+An alternative solution is that CometBFT, itself, using the implementation of the `dump` command, whichever 
+is chosen at the time, creates or asks the application to create snapshot exports. 
+
+**Export file consistency**
 
 The same caveat of making sure the data exported is not corrupted by concurrent reads and writes applies here as well. If we decide to simply export
-snapshots into a compressed file, we need to make sure that we do not return files that are not already written. 
+snapshots into a compressed file, we need to make sure that we do not return files that with ongoing writes.
 An alternative is for CometBFT to provide a Snapshot export store with snapshot isolation. This store would essentially be pruned as soon as a new snapshot is written,
 thus not taking too much more space.
 
+If it is the application that exports the snapshots, it is something the application developer has to be aware of.
+
+#### Syncing a node using local snapshots
+
+On startup, if `local_sync` is set to `true`, CometBFT will try to look for an existing snapshot at the path
+given by the operator. The snapshots will be loaded and state restored as if they came from a peer in the current implementation. 
+
+Note that, if it is not CometBFT that created the snapshot export from the data retrieved via ABCI (a combination of `ListSnapshots` and `LoadChunks`),
+CometBFT might not be aware of how the snapshot was exported, and needs to ask the application to restore the snapshot.
+
+If a snapshot was created using option 1 from the previous section, or the export format is known to CometBFT (like `tar, gzip` etc.), 
+CometBFT can extract the snapshot itself, and offer it to the application via `RequestOfferSnapshot` without any API changes. 
+
+If CometBFT is not the one who created the exported file, we introduce a new ABCI call `UnpackSnapshot` 
+to send the exported snapshot as a blob of bytes to the application, which uncompresses it, installs it 
+and responds whether the snapshot is accepted and the chunk application has passed. 
+
+
+* **Request**:
+
+    | Name            | Type    | Description                                 | Field Number |
+    |-----------------|---------|---------------------------------------------|--------------|
+    | exportedSnapshot| [] byte | Snapshot export created by the application. | 1            |
```

Comment:
So is Comet that saves the snapshot to a file?

Reply:
In this case no.. This call is reserved for reading an exported snapshot that comet is just reading in as bytes and sending to the app for parsing.